shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DMLStatement.g4

```diff
@@ -101,15 +101,19 @@ multipleTableNames
     ;
 
 select 
-    : unionClause
+    : selectSubquery forUpdateClause?
+    ;
+
+selectSubquery
+    : (selectClause | selectClause ((UNION ALL? | INTERSECT | MINUS) selectSubquery)+ | LP_ selectSubquery RP_) orderByClause? rowLimitingClause
```

Comment:
`(selectClause | ` should be `(queryBlock | `, right?

Reply:
Yes @wgy8283335, `selectClause` refers to `queryBlock`, I'll rename `selectClause` to `queryBlock`. 

When we specify `selectSubquery` as:
```
selectSubquery
    : (queryBlock | selectSubquery ((UNION ALL? | INTERSECT | MINUS) selectSubquery)+ | LP_ selectSubquery RP_) orderByClause? rowLimitingClause
    ;
```
ANTLR gives an error: `selectSubquery` is mutually left-recursive. 
![SelectSubquery-MutuallyLeftRecursive](https://user-images.githubusercontent.com/48581379/118752054-0be7d700-b880-11eb-869c-f63824d33da0.png)

I think error is coming because of this one: `selectSubquery ((UNION ALL? | INTERSECT | MINUS) selectSubquery)+`

Reply:
To resolve the mutually left-recursive error, I tried with this definition:

```
select 
    : selectSubquery forUpdateClause?
    ;

selectSubquery
    : (queryBlock | selectUnionClause | parenthesisSelectSubquery) orderByClause? rowLimitingClause
    ;

selectUnionClause
    : ((queryBlock | parenthesisSelectSubquery) orderByClause? rowLimitingClause) ((UNION ALL? | INTERSECT | MINUS) selectSubquery)+
    ;

parenthesisSelectSubquery
    : LP_ selectSubquery RP_
    ;
```
It worked without that error. Because of mutually left-recursive error, I didn't add `selectUnionClause` in `selectUnionClause`'s definition and only specified `(queryBlock | parenthesisSelectSubquery)`.